packages/next/src/server/render.tsx

```diff
@@ -1348,7 +1348,7 @@ export async function renderToHTMLImpl(
       (initialStream: ReactReadableStream, suffix?: string) => {
         return continueFizzStream(initialStream, {
           suffix,
-          inlinedDataStream: serverComponentsInlinedTransformStream?.readable,
+          inlinedDataStream: (serverComponentsInlinedTransformStream as any)?.readable,
```

Comment:
side-effect from upgrading to ts 5.4. didn't look into the root cause

Reply:
can revert back to 5.2.2 and do some `(ts.ModuleKind as any).Preserve` in `writeConfigurationDefaults`, although I'm not sure if the tests will pass then since `Preserve` isn't recognized and might throw a build-error in the test